### PR TITLE
Fix avoid_positional_boolean_parameters for implementing classes

### LINKED ISSUE #786: avoid_positional_boolean_parameters should ignore overridden methods
In the following code, `avoid_positional_boolean_parameters` is reported even in the `Bar` class. I think the rule should only warn the original class that defines the signature.

```dart
abstract class Foo {
  void foo(bool positionalBoolean);
}

class Bar implements Foo {
  @override
  void foo(bool positionalBoolean) {}
}
```

Interestingly, when I change `implements` to `extends`, the warning goes away.

### PR DESCRIPTION
**Pull request by srawlins**
Fix #786

### CODE REVIEW

File: lib/src/rules/avoid_positional_boolean_parameters.dart [hunk 1]
```diff
@@ -102,4 +113,17 @@ class _Visitor extends SimpleAstVisitor<void> {
       DartTypeUtilities.implementsInterface(
           node.identifier.bestType, 'bool', 'dart.core') &&
       !node.isNamed;
+
+  bool _isOverridingMember(Element member) {
```
Comment (srawlins) [inline lib/src/rules/avoid_positional_boolean_parameters.dart]: This jazz is largely copied from `lib/src/rules/annotate_overrides.dart`. I'd extract it to be a static utility, but it uses `manager`, instantiated in `visitCompilationUnit`. I wonder if this `manager` might become useful enough to create a parent visitor. `abstract class InheritanceAwareAstVisitor extends AstVisitor` or something...

Comment (pq) [reply lib/src/rules/avoid_positional_boolean_parameters.dart]: @bwilkerson: thoughts on that?

File: lib/src/rules/avoid_positional_boolean_parameters.dart [hunk 2]
```diff
@@ -60,8 +60,18 @@ class AvoidPositionalBooleanParameters extends LintRule
 class _Visitor extends SimpleAstVisitor<void> {
   final LintRule rule;
 
+  InheritanceManager manager;
+
   _Visitor(this.rule);
 
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    LibraryElement library = node == null
```
Comment (bwilkerson) [inline lib/src/rules/avoid_positional_boolean_parameters.dart]: Visit methods should never be invoked with a null argument, so we shouldn't need this test. Did you run into an actual problem, or were you just being defensive?

File: lib/src/rules/avoid_positional_boolean_parameters.dart (hunk 1 above)
Comment (bwilkerson) [reply lib/src/rules/avoid_positional_boolean_parameters.dart]: We haven't decided yet whether InheritanceManager will be part of the public API or whether we'll have a different mechanism for answering questions like this. I'd postpone linter changes until the analyzer API is finalized.

File: lib/src/rules/avoid_positional_boolean_parameters.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/rules/avoid_positional_boolean_parameters.dart]: Literally just copied from `lib/src/rules/annotate_overrides.dart`. Good point. Deleting

File: lib/src/rules/avoid_positional_boolean_parameters.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/rules/avoid_positional_boolean_parameters.dart]: SG

**Pull request merged**